pkg/core/store/badger/workflow.go

```diff
@@ -0,0 +1,171 @@
+package badger
+
+import (
+	"context"
+	"log"
+
+	"github.com/fuseml/fuseml-core/gen/workflow"
+	"github.com/fuseml/fuseml-core/pkg/domain"
+	"github.com/timshannon/badgerhold/v3"
+)
+
+// storableWorkflow holds the workflow and the codesets assigned to it
+type storableWorkflow struct {
+	// Workflow assigned to the codeset
+	Workflow *workflow.Workflow
+	// AssignedCodeset holds codesets assigned to the workflow and its hookID
+	AssignedCodesets []*domain.AssignedCodeset
+}
+
+type WorkflowStore struct {
+	store *badgerhold.Store
+}
+
+func NewWorkflowStore(options badgerhold.Options) *WorkflowStore {
+	store, err := badgerhold.Open(options)
```

Comment:
BTW when is the time to `badgerhold.Close`?

Reply:
I believe the best place for it is at `cmd/fuseml_core/main.go` just need to figure out the best way to get it from wire.

I did not spend time on it as the idea of that PR is to evaluate the storage option.

Reply:
Would it be too bad to open and close the file for every write? I actually think we do not expect huge I/O traffic ...

Reply:
I agree, but I am not sure about the advantages when doing that compared to one open/close.